### ISSUE TITLE
tensorflow-macos cannot train model

### ISSUE THREAD

**Issue by Yuri-Su** [tensorflow/tensorflow#53093]
**System information**

- OS Platform and Distribution):macOS 12.1
- TensorFlow installed from (source or binary):tensorflow-macos pip
- TensorFlow version (use command below):2.6
- Python version:3.8
- Bazel version (if compiling from source):
- GPU model and memory: AMD RX6600 8G

**Unable to train model**
```
import tensorflow as tf

mnist = tf.keras.datasets.mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0
w, h = 28, 28

x_train = x_train.reshape(x_train.shape[0], w, h, 1)
x_test = x_test.reshape(x_test.shape[0], w, h, 1)

model = tf.keras.models.Sequential([
    tf.keras.layers.Conv2D(filters=64, kernel_size=2, padding='same', activation='relu', input_shape=(28, 28, 1)),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(512, activation=tf.nn.relu),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test, y_test)
```

**Run Information**
```
2021-11-17 16:07:45.954518: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-11-17 16:07:45.955017: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:305] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2021-11-17 16:07:45.955191: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:271] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: METAL, pci bus id: <undefined>)
Metal device set to: AMD Radeon RX 6600

systemMemory: 16.00 GB
maxCacheSize: 3.99 GB

2021-11-17 16:07:46.368598: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
Epoch 1/5
2021-11-17 16:07:46.584915: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:112] Plugin optimizer for device_type GPU is enabled.
2021-11-17 16:07:46.623 python[4372:114284] -[MPSGraph adamUpdateWithLearningRateTensor:beta1Tensor:beta2Tensor:epsilonTensor:beta1PowerTensor:beta2PowerTensor:valuesTensor:momentumTensor:velocityTensor:gradientTensor:name:]: unrecognized selector sent to instance 0x600007d83720

Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

**Comment by mohantym** [comment]
Hi @Yuri-Su ! Could not find any issue  in 2.7 ,It was throwing AlreadyExistsError in 2.6 . Attaching [Gist](https://colab.sandbox.google.com/gist/mohantym/042ff221befd4824b51563e8b7889695/github_53093.ipynb) fore reference. 
For further assistance. 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by Yuri-Su** [comment]
This problem was solved in the latest version 2.7, the metal version is 0.3.0. But the adam optimizer still has problems

**Comment by jzhangc** [comment]
I have the same problem: Adam optimizer would not work. Although SGD works, it does not actually optimize anything, i.e. loss does not improve over epochs.
Intel iMac 27 (2020, AMD Radeon Pro 5500 XT)
`[MPSGraph adamUpdateWithLearningRateTensor:beta1Tensor:beta2Tensor:epsilonTensor:beta1PowerTensor:beta2PowerTensor:valuesTensor:momentumTensor:velocityTensor:gradientTensor:name:]: unrecognized selector sent to instance 0x60002836b9c0`
Versions are as following:
```
tensorflow-macos: tensorflow_macos-2.8.0-cp38-cp38-macosx_11_0_x86_64
tensor flow-metal: tensorflow_metal-0.4.0-cp38-cp38-macosx_11_0_x86_64
python: 3.8
macOS: 12.3
```
I have also tried to roll back tensorflow-macos to 2.7 and tensorflow-metal to 0.3.0, as well as mix&match them. Nothing worked.
Meanwhile, my M1 MacBook Pro works without any problem with the following:
```
tensorflow-macos: tensorflow_macos-2.8.0-cp39-cp39-macosx_11_0_arm64
tensorflow-metal: tensorflow_metal-0.4.0-cp39-cp39-macosx_11_0_arm64
python: 3.9
macOS: 12.3
```
Since intel Mac does not have cp39 version for tensorflow-metal, I have no way to test if it is a TF2 problem, metal problem or a macOS 12.3 problem.
